src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/JinaAIEmbeddingClient.java

```diff
@@ -54,7 +57,53 @@ Uni<EmbeddingResponse> embed(
 
     @ClientExceptionMapper
     static RuntimeException mapException(jakarta.ws.rs.core.Response response) {
-      return HttpResponseErrorMessageMapper.getDefaultException(response);
+      String errorMessage = getErrorMessage(response);
+      return HttpResponseErrorMessageMapper.mapToAPIException(
+          ProviderConstants.JINA_AI, response, errorMessage);
+    }
+
+    /**
+     * Extract the error message from the response body. The example response body is:
+     *
+     * <pre>
+     * {
+     *    "detail": "ValidationError(model='TextDoc', errors=[{'loc': ('text',), 'msg': 'Single text cannot exceed 8192 tokens. 10454 tokens given.', 'type': 'value_error'}])"
+     * }
+     * </pre>
+     *
+     * <pre>
+     *     {"detail":"Failed to authenticate with the provided api key."}
+     * </pre>
+     *
+     * @param response The response body as a String.
+     * @return The error message extracted from the response body.
+     */
+    private static String getErrorMessage(jakarta.ws.rs.core.Response response) {
+      String responseBody = response.readEntity(String.class);
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode rootNode = mapper.readTree(responseBody);
+        // Extract the "detail" node
+        JsonNode detailNode = rootNode.path("detail");
+        // Assuming the error message is within the "msg" part of the detail
+        if (!detailNode.isMissingNode()) {
+          String detailText = detailNode.asText();
+          // Use regex to extract the message part from the detail text
+          java.util.regex.Pattern pattern = java.util.regex.Pattern.compile("'msg':\\s*'([^']*)'");
```

Comment:
Minor detail but is it possible that the `'msg': '...'` will contain an escaped single quote like `'msg': 'Single text can\'t exceed 8192 tokens'`? If so, likely worthwhile to handle that here, but I haven't taken a look at Jina AI's docs so I wouldn't know.